### ISSUE TITLE
Comments get decolorized in some newly created projects

### ISSUE THREAD

**Issue by psfinaki** [dotnet/fsharp#16886]
**Repro steps**

1. First create a new F# Console App
2. Then create a new F# Console Application(.NET Framework)
3. Observe Program.fs

**Expected behavior**

All good, nothing special happens.

**Actual behavior**

Comments go gray:

![945492ac-8132-4236-8f8c-60555a282950](https://github.com/dotnet/fsharp/assets/5451366/65672c2f-0c55-4552-80bc-279cb617798c)

**Known workarounds**

Start editing the doc, things go back to normal.

**Related information**

17.10.0 Preview 3.0 [34714.165.main], X64 and ARM64.

**Comment by psfinaki** [comment]
The ticket above in a copy from an internal backlog, we cannot repro this fully yet, but I can see something similar (colors disappearing right away without a few minutes' lag):

https://github.com/dotnet/fsharp/assets/5451366/450aa39f-bdd9-4233-a8f6-73a380680057

**Comment by psfinaki** [comment]
Need to see if it's even an F# thing. Not many templates have comments there right away, so it's not clear yet.

**Comment by vzarytovskii** [comment]
I know Petr P. had the same issue sometimes in his setup, but I didn't and we never figured out. Right now colouring (roughly) work as follows:
1. Roslyn asks us about both syntactic and semantic highlighting for specific ranges in document
2. We just return what's on these ranges
3. Roslyn/vs colours it.

What I suspect can be happening:
a. We return incorrect type for some range at some point for some reason. 
b. Roslyn or some other "extension" for some reason resets it at some point.
c. There's some weird set of dogfood options for some accounts which reset colouring settings themselves for a brief period.

I don't see any other semantic colouring broken in the repros, as well as it's reported, that only new projects are affected, it's not critical, but we better investigate it since it can (theoretically) be affecting something else.

**Comment by majocha** [comment]
FWIW I've seen this quite often at some point when opening existing slns, but not recently. The only classification affected was the comment, everything else was normal, so I assumed it's some bug on VS side.

**Comment by vzarytovskii** [comment]
> so I assumed it's some bug on VS side.

Yeah, that what I was thinking as well. I have never seen it in my setup. Like not a single time :)

**Comment by majocha** [comment]
Oops, just right now. Opening FSharp.sln with some preopened documents.
![image](https://github.com/dotnet/fsharp/assets/1760221/52447ad8-567a-437f-9962-922582394f10)

This is clearly the default plain text classifictation. Only the current document is affected when opening VS. The rest look fine. As soon as I do some edit in the file it gets back to normal. So this is minimal impact IMV.

Should look like this in my case:

![image](https://github.com/dotnet/fsharp/assets/1760221/7e8fc59d-52f7-4510-8590-6e3ef837da78)
